src/pip/_internal/pep425tags.py

```diff
@@ -117,7 +117,9 @@ def get_abi_tag():
             d = 'd'
         if get_flag('WITH_PYMALLOC',
                     lambda: impl == 'cp',
-                    warn=(impl == 'cp')):
+                    warn=(impl == 'cp' and
+                          sys.version_info < (3, 8))) \
+                and sys.version_info < (3, 8):
```

Comment:
Question: is there a reason the `sys.version_info < (3, 8)` has to go after the call to `get_flag()`? It seems like this would be simpler with it before because then the `sys.version_info < (3, 8)` wouldn't need to be duplicated.

Reply:
At a quick glance, that sounds like a reasonable tidy-up. A follow-up PR (as I just merged this as it stands) to clean this up would seem reasonable to me.

Reply:
Sounds good. (And the same type of change can be made in the following conditional as well.)

Reply:
I agree, I was just following the example of the other version check below.

Reply:
I just filed this as PR #6889.